### PR TITLE
Added ability to print version info

### LINKED ISSUE #306: Add version option to mepo?
Should we add the ability to do `mepo --version`?

### PR DESCRIPTION
**Pull request by pchakraborty**
This resolves #306 and #301

### CODE REVIEW

**Review by mathomp4 — APPROVED** [review]
Well that is not how I thought this would be done.

Should we add a unit test for this? I suppose it can't really break.

**Comment by pchakraborty** [comment]
This way, there is only one version number, in `pyproject.toml`.

For the `develop` branch, I suppose I can bump the version number and test against that.

**Comment by pchakraborty** [comment]
@mathomp4 On second thought, it's probably best to test against the last released version number.

File: requirements.lock [hunk 1]
```diff
@@ -10,5 +10,5 @@
 -e file:.
 colorama==0.4.6
     # via mepo
-pyyaml==6.0.1
+pyyaml==6.0.2
```
Comment (mathomp4) [inline requirements.lock:13]: This is interesting. Spack doesn't seem to see the version of pyyaml! I'll ask on the slack...

Comment (pchakraborty) [reply requirements.lock]: We "shouldn't" need to version control the lock files. They are to be generated by `rye` during `rye sync`. I removed them, but they came back while merging in the changes from `main`.

Comment (pchakraborty) [reply requirements.lock]: @mathomp4 Removed dependencies on `requirements**.lock`. This should be good to go.

**Review by pchakraborty — COMMENTED** [review]
Removed `lock` files from version control. Earlier `requirements-dev.lock` was used by `pip` to install dependencies. Now we use `rye sync` which reads `toml`, and creates the `lock` files. `pyyaml` in `pyproject.toml` remains at `>=6.0.1`.

**Review by mathomp4 — APPROVED** [review]

**Pull request merged**